scripts/data/src/org/akvo/gae/remoteapi/PathToIdMigration.java

```diff
@@ -0,0 +1,116 @@
+package org.akvo.gae.remoteapi;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Query;
+
+public class PathToIdMigration implements Process {
+
+    private static final String SURVEY_GROUP = "SurveyGroup";
+    private static final String SURVEY = "Survey";
+    private static final String USER_AUTHORIZATION = "UserAuthorization";
+
+    @Override
+    public void execute(DatastoreService ds, String[] args) throws Exception {
+
+        final List<Entity> surveyGroups = ds.prepare(new Query(SURVEY_GROUP))
+                .asList(FetchOptions.Builder.withDefaults());
+        final List<Entity> surveys = ds.prepare(new Query(SURVEY)).asList(
+                FetchOptions.Builder.withDefaults());
+        final List<Entity> userAuthorizations = ds.prepare(
+                new Query(USER_AUTHORIZATION)).asList(
+                FetchOptions.Builder.withDefaults());
+
+        Map<Long, Long> idToParentId = new HashMap<>();
+        Map<String, Long> pathToId = new HashMap<>();
+
+        for (Entity entity : surveyGroups) {
+            // Ensure root folder parentId == 0
+            Long entityId = entity.getKey().getId();
+
+            if (entity.getProperty("parentId") == null) {
+                entity.setProperty("parentId", 0L);
+            }
+            idToParentId.put(entityId, (Long) entity.getProperty("parentId"));
+
+            String path = (String) entity.getProperty("path");
+            if (path != null) {
+                pathToId.put(path, entityId);
```

Comment:
@jonase Perhaps we can also log here if the path is `null` so that we know which `SurveyGroup` entities are not going to be considered.